x/data/server/features/define_resolver.feature

```diff
@@ -1,15 +1,12 @@
 Feature: Define Resolver
 
-  Background:
-    Given the resolver url "https://foo.bar"
+  Rule: the resolver is defined with no unique constraints on the URL
 
-  Rule: the resolver is defined if the resolver url is unique
+    Scenario: the url is unique
```

Comment:
```suggestion
    Scenario: a resolver for a new URL is registered
```

Reply:
In general, I was thinking the scenario would be short and to the point and the rule would be more descriptive. Given the rule is "the resolver is defined with no unique constraints on the URL", the two scenarios are "the url is unique" and "the url is not unique". This is also how the BDD books write scenarios. For example,

```
Rule: Any visitor to the website can place a customer-collection order

  Scenario: Unauthenticated customer chooses to collect order
    Given the customer is unauthenticated
    When they choose to collect their order
    Then they should be asked to supply contact details

  Scenario: Authenticated customer chooses to collect order
    Given the customer is authenticated
    When they choose to collect their order
    Then they should be asked to confirm contact details
```

Reply:
This also produces the test logs:

```
=== RUN   TestDefineResolver
=== RUN   TestDefineResolver/Define_Resolver
=== RUN   TestDefineResolver/Define_Resolver/a_resolver_for_a_new_url_is_registered
=== RUN   TestDefineResolver/Define_Resolver/a_resolver_is_registered_with_a_url_that_is_already_registered
```

Versus the current test logs:

```
=== RUN   TestDefineResolver
=== RUN   TestDefineResolver/Define_Resolver
=== RUN   TestDefineResolver/Define_Resolver/the_url_is_unique
=== RUN   TestDefineResolver/Define_Resolver/the_url_is_not_unique
```

Reply:
We are also "defining" a resolver here, not "registering" data to the resolver.